### PR TITLE
Make sure blob columns are correctly converted as parameters

### LINKED ISSUE #173: Database conversion: hangs forever
### Steps to reproduce
1. Create an empty PostgreSQL database
2. Perform a database conversion: `./occ db:convert-type --password <password> --all-apps --clear-schema pgsql <username> <hostname> <database>`
### Expected behaviour

A working database conversion.
### Actual behaviour

The conversion hangs forever at `oc_cards`:

```
[snipped]
oc_activity
 4948/4948 [============================] 100%
oc_activity_mq
    0/0 [============================]   0%
oc_addressbooks
 8/8 [============================] 100%
oc_cards
  330/1162 [=======>--------------------]  28%
```
### Server configuration

**Operating system**: Arch Linux

**Web server:** Nginx

**Database:** SQLite

**PHP version:** 7.0.7

**Nextcloud version:** 9.0.50

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated, starting from ownCloud 3.0

**Where did you install Nextcloud from:** archive

**Signing status (Nextcloud 9.0 and above):** 

```
No errors have been found.
```

**List of activated apps:**

```
Enabled:
  - activity: 2.2.1
  - admin_audit: 1.0.0
  - calendar: 1.2.2
  - comments: 0.2
  - contacts: 1.3.1.0
  - dav: 0.1.6
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_pdfviewer: 0.8.1
  - files_sharing: 0.9.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - notes: true
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - tasks: true
  - templateeditor: 0.1
  - updatenotification: 0.1.0
Disabled:
  - encryption
  - external
  - files_external
  - files_texteditor
  - user_external
  - user_ldap
```

**Are you using encryption:** no

### LINKED ISSUE #3536: DB Migration from SQLite to PGSQL fails
### Steps to reproduce

1. Create an empty PostgreSQL database
2. Perform a database conversion: sudo -u www occ db:convert-type --all-apps  pgsql <username> <hostname> <database>

### Expected behaviour

A working database conversion.

### Actual behaviour
Migration process fails with error message "SQLSTATE[22P02]: Invalid text representation: 7 FEHLER:  ungültige Eingabesyntax für Typ bytea" at oc_cards migration 

    0 [>---------------------------]oc_file_locks
    80/80 [============================] 100%oc_comments
    0 [>---------------------------]oc_comments_read_markers
    0 [>---------------------------]oc_credentials
    2/2 [============================] 100%oc_admin_sections
    5/5 [============================] 100%oc_admin_settings
   10/10 [============================] 100%oc_accounts
    3/3 [============================] 100%oc_federated_reshares
    0 [>---------------------------]oc_addressbooks
   3/3 [============================] 100%oc_cards
    65/412 [====>-----------------------]  15%

     [Doctrine\DBAL\Exception\DriverException]
    An exception occurred while executing 'INSERT INTO "oc_cards" ("id", "addressbookid", "carddata", "uri", "lastmodified", "etag", "size") VALUES(?, ?, ?, ?, ?, ?, ?)' with params [ <SNIP/>

### Server configuration
**Operating system**: FreeBSD 11.0

**Web server:** NGINX

**Database:** SQLITE3 to Postgresql 9.3

**PHP version:** 5.6

**Nextcloud version:** 11.0.1 (stable)

**Updated from an older Nextcloud/ownCloud or fresh install:** MIgrated from Owncloud 9.1

**Where did you install Nextcloud from:** FreeBSD ports

**Signing status:**
<details>
<summary>Signing status</summary>

all files failed
</details>

**List of activated apps:**
<details>
Enabled:
  - calendar: 1.5.0
  - comments: 1.1.0
  - contacts: 1.5.3
  - dav: 1.1.1
  - federatedfilesharing: 1.1.1
  - federation: 1.1.1
  - files: 1.6.1
  - files_external: 1.1.2
  - files_pdfviewer: 1.0.1
  - files_sharing: 1.1.1
  - files_texteditor: 2.2
  - files_videoplayer: 1.0.0
  - gallery: 16.0.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - nextcloud_announcements: 1.0
  - notifications: 1.0.1
  - password_policy: 1.1.0
  - provisioning_api: 1.1.0
  - serverinfo: 1.1.1
  - sharebymail: 1.0.1
  - systemtags: 1.1.3
  - theming: 1.1.1
  - twofactor_backupcodes: 1.0.0
  - updatenotification: 1.1.1
  - workflowengine: 1.1.1
Disabled:
  - activity
  - admin_audit
  - encryption
  - external
  - files_accesscontrol
  - files_automatedtagging
  - files_retention
  - files_trashbin
  - files_versions
  - firstrunwizard
  - spreed
  - survey_client
  - templateeditor
  - user_external
  - user_ldap
  - user_saml

</details>

**The content of config/config.php:**
<details>
<summary>Config report</summary>

{
    "system": {
        "instanceid": "oc8cd111a9dc",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "**REMOVED SENSITIVE VALUE***",
            "**REMOVED SENSITIVE VALUE***",
            "**REMOVED SENSITIVE VALUE***",
            "**REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "\/usr\/local\/www\/nextcloud\/data",
        "dbtype": "sqlite3",
        "version": "11.0.1.2",
        "installed": true,
        "forcessl": true,
        "theme": "",
        "maintenance": false,
        "secret": "***REMOVED SENSITIVE VALUE***",
        "appstore.experimental.enabled": true,
        "loglevel": 2,
        "htaccess.RewriteBase": "\/"
    }
}

</details>

**Are you using external storage, if yes which one:** smb

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### PR DESCRIPTION
**Pull request by nickvergessen**
Fix #173 
Fix #3536

### CODE REVIEW

**Comment by nickvergessen** [comment]
I'd like to backport this to 11, so we can tell people that they can migrate away from sqlite on 11+ at least, @karlitschek

**Comment by karlitschek** [comment]
nice. please backport 👍

File: core/Command/Db/ConvertType.php [hunk 1]
```diff
@@ -313,6 +317,25 @@ protected function copyTable(Connection $fromDB, Connection $toDB, $table, Input
 		$progress->finish();
 	}
 
+	protected function getColumnType($table, $column) {
+		if (isset($this->columnTypes[$table][$column])) {
+			return $this->columnTypes[$table][$column];
+		}
+
+		$prefix = $this->config->getSystemValue('dbtableprefix', 'oc_');
+		$this->columnTypes[$table][$column] = null;
```
Comment (icewind1991) [inline core/Command/Db/ConvertType.php]: setting this makes no difference, `isset` will also return false if the value is null.

**Review by icewind1991 — CHANGES_REQUESTED** [review]

File: core/Command/Db/ConvertType.php (hunk 1 above)
Comment (nickvergessen) [inline core/Command/Db/ConvertType.php]: Right, I used a different value before...

Comment (nickvergessen) [inline core/Command/Db/ConvertType.php]: Fixed

**Comment by rullzer** [comment]
Is there anyway to have tests for this?

**Comment by nickvergessen** [comment]
The problem is, I can create a test whether the `isset($data[$table][$colum])` is true, not sure it if makes sense. We also need to maintain the list manually...

**Review by rullzer — APPROVED** [review]
ok lets do this

**Review by icewind1991 — APPROVED** [review]

**Pull request merged**